### PR TITLE
Fix bed packet

### LINKED ISSUE #1398: Find more reliable ways to know if the bot is sleeping and add tests
The 'bed' event has been removed from Minecraft [since 1.14](https://wiki.vg/Protocol_History#19w08a). That event was used here: https://github.com/PrismarineJS/mineflayer/blob/1b1907d48d4eaf5be990a1b53dd17cac18b11c53/lib/plugins/entities.js#L106-L111

And so `bot.isSleeping` never gets updated:
https://github.com/PrismarineJS/mineflayer/blob/1b1907d48d4eaf5be990a1b53dd17cac18b11c53/lib/plugins/bed.js#L166-L171

Which then blocks the players from running `bot.wake()`
https://github.com/PrismarineJS/mineflayer/blob/1b1907d48d4eaf5be990a1b53dd17cac18b11c53/lib/plugins/bed.js#L72-L83

I think we should find a reliable way to detect if the bot is sleeping. 
After a bit of testing I think this would be one option: Listening for  [Entity Metadata event](https://wiki.vg/Protocol#Entity_Metadata):
```json
{
  "entityId": 41,
  "metadata": [
    {
      "key": 6,
      "type": 18,
      "value": 2
    },
    {
      "key": 13,
      "type": 10,
      "value": {
        "x": -273,
        "z": 85,
        "y": 4
      }
    }
  ]
}
```

I think the ['wake' event](https://github.com/PrismarineJS/mineflayer/blob/1b1907d48d4eaf5be990a1b53dd17cac18b11c53/lib/plugins/entities.js#L113-L118) is being emitted correctly, as I think there have not been modifications to that part of the protocol. But we definetely need to add external tests for all of this. I'll work on this when I have some time :smile:

### PR DESCRIPTION
**Pull request by imharvol**
Closes #1398 and #1400

### CODE REVIEW

**Comment by imharvol** [comment]
I'm not entirely sure about this:
https://github.com/PrismarineJS/mineflayer/blob/6d54b7bfc5e49e9988661f7f7b2620dd71b8af96/lib/plugins/entities.js#L106-L111

Should I create a feature in features.json that contains which versions 'bed' packet works and then put that code inside?
```js
if (bot.supportFeature('sleepPacket', bot.version)) {
  bot._client.on('bed', (packet) => { 
    // use bed 
    const entity = fetchEntity(packet.entityId) 
    entity.position.set(packet.location.x, packet.location.y, packet.location.z) 
    bot.emit('entitySleep', entity) 
  }) 
}

File: lib/plugins/entities.js [hunk 1]
```diff
@@ -359,6 +359,13 @@ function inject (bot, { version }) {
       bot.emit('itemDrop', entity)
     }
 
+    const pose = packet.metadata.find(e => e.type === 18)
+    if (pose) {
+      if (pose.value === 2) {
+        bot.emit('entitySleep', entity)
+      }
+    }
```
Comment (NotSugden) [inline lib/plugins/entities.js:367]:
```suggestion
    if (pose && pose.value === 2) {
      bot.emit('entitySleep', entity)
    }
```

**Comment by imharvol** [comment]
It's not quite clear to me why test are failing. Anyone knows?

File: test/externalTests/bed.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        bot.chat(`/setblock ${bedPos1.toArray().join(' ')} ${bedItem.name}[part=foot]`)
+        bot.chat(`/setblock ${bedPos2.toArray().join(' ')} ${bedItem.name}[part=head]`)
+      }
+      setTimeout(cb, 1000) // Enough time for the game-time and the block to be updated
+    },
+    (cb) => { // Sleep
+      assert(!bot.isSleeping)
+      bot.sleep(bot.blockAt(bedPos1), (err) => {
+        assert.ifError(err)
+        setTimeout(cb, 500) // Enough time for bot.isSleeping to be updated
```
Comment (rom1504) [inline test/externalTests/bed.js:29]: this will create flaky tests, can you use an event instead ?

**Review by rom1504 — CHANGES_REQUESTED** [review]

File: test/externalTests/bed.js (hunk 2 above)
Comment (imharvol) [reply test/externalTests/bed.js]: Pretty dumb of me not to use 'sleep' and 'wake' events so they're tested too

**Review by rom1504 — APPROVED** [review]

**Pull request merged**